swarm/src/behaviour/either.rs

```diff
@@ -237,6 +236,7 @@ where
     }
 }
 
+#[allow(deprecated)]
 impl<TEvent, TBehaviourLeft, TBehaviourRight> NetworkBehaviourEventProcess<TEvent>
     for Either<TBehaviourLeft, TBehaviourRight>
 where
```

Comment:
What is the usage case for `NetworkBehaviourEventProcess` on `Either` (or `Toggle` below)?
As far as I understood it, the purpose of `NetworkBehaviourEventProcess` is only to be used in combination with the `NetworkBehaviour` derive macro.

Reply:
I guess when you use `Either` or `Toggle` when building a nested `NetworkBehaviour`. See for example:

https://github.com/libp2p/rust-libp2p/blob/a4110a2b6939d5b460f11df6dd158308f517becf/swarm-derive/tests/test.rs#L354-L369

Reply:
Here with `NetworkBehaviourEventProcess`:

https://github.com/libp2p/rust-libp2p/blob/2f2b7cbec127a48d869d87ad52d69a4c4c29ddb5/swarm-derive/tests/test.rs#L328-L352

Reply:
But you wouldn't call the `NetworkBehaviourProcess` implementation of an inner behaviours, no? Instead you implement `NetworkBehaviourProcess` on the parent (in the above case in `Foo`) and handle the event there. Why would anyone want to inject the event back into an inner behaviour?

None of our behaviours currently implement `NetworkBehaviourProcess it, so the below trait bound for it would not apply anyway:
https://github.com/libp2p/rust-libp2p/blob/a4110a2b6939d5b460f11df6dd158308f517becf/swarm/src/behaviour/toggle.rs#L237-L239
So `libp2p::ping::Ping` does not implement `NetworkBehaviourProcess`, therefore `Toggle<libp2p::ping::Ping>` will also not implement it.

Reply:
Unless I am missing something, there was never really a use-case for `NetworkBehaviourEventProcess` on those two behaviours in the first place. However, that's not really a concern of this PR. 
So I guess I am fine with for now with keeping it the way it is,  since this trait will be remove anyway in the next release.